lib/iris/tests/runner/_runner.py

```diff
@@ -129,35 +129,26 @@ def run(self):
                     "WARNING: Gallery path %s does not exist." % (gallery_path)
                 )
         if not tests:
-            tests.append("iris.tests")
-
-        regexp_pat = r"--match=^([Tt]est(?![Mm]ixin)|[Ss]ystem)"
```

Comment:
Might be worth renaming [`test_constraints.TestMixin`](https://github.com/SciTools/iris/blob/c047007228bafc7f38d78c0d2b5b54d32b387cb9/lib/iris/tests/test_constraints.py#L94)?

It doesn't contain any tests so won't be run, but we don't want to mislead users into thinking that it's OK to start a Mixin name with `Test`. If I understand correctly the only remaining protection against `Mixin` tests being run is that they don't start with `Test`. Please correct me if I'm wrong.

Reply:
Yes.  See https://github.com/SciTools/iris/pull/4249#discussion_r674632721.